### ISSUE TITLE
1D or 2D pseudo counter cannot be created with more than one pseudo counter role / axis

### ISSUE THREAD

**Issue by tox1c90** [sardana-org/sardana#1440]
I try to create 1D and 2D pseudo counter controllers which integrate multiple different ROIs along one axis of an image or treating the whole image differently depending on the axis of the pseudo counter.

In case of a 0D pseudo counter, this works perfectly by defining multiple pseudo counter roles and evaluating the axis variable within the calculation function.

However, when I try to define more than one pseudo counter roles for a 1D or 2D controller, I get an error. See the following simple example which just multiplies the image by the number of the axis:

```
class dummy2DMultipleAxisController(PseudoCounterController):

    counter_roles        = ('I',)
    pseudo_counter_roles = ('O1','O2','O3','O4',)
    
    def __init__(self, inst, props, *args, **kwargs):
        PseudoCounterController.__init__(self,inst,props, *args, **kwargs)
        
    def GetAxisAttributes(self, axis):
        axis_attrs = PseudoCounterController.GetAxisAttributes(self, axis)
        axis_attrs = dict(axis_attrs)
        axis_attrs['Value'][Type] = ((float, ), )
        axis_attrs['Value'][MaxDimSize] = (1024,1024)
        return axis_attrs
    
    def Calc(self, axis, counters):
        image = counters[0]        
        return axis*image
```

Doing `defctrl dummy2DMultipleAxisController dummy2DMultipleAxisCtrl I=dummy2D O1=dummy2Dx1 O2=dummy2Dx2 O3=dummy2Dx3 O4=dummy2Dx4` with a `dummy2D` counter as input, throws an error:

`An error occurred while running defctrl:
API_AttrNotFound: value attribute not found`

Only the first pseudo counter dummy2Dx1 is created properly. The second one is created but as a scalar, and trying to add it to the measurement group leads to `Attribute Value is not supported by device pc/dummy2dmultipleaxisctrl/2`.

Am I doing something wrong here, or is this a bug?

**Comment by reszelaz** [comment]
Hi Martin,
First, many thanks for this comprehensive report!
I have just tested your example and I get the same result. This looks like a bug. To be honest I have never tried pseudo counters with multiple pseudo axes of 1D or 2D shape. 
Probably you have already discovered that having only one pseudo axis works for 1D and 2D shape. 
Meanwhile we find some time to look into this deeper would it work for you as a workaround to have multiple controller classes of one axis only?

**Comment by tox1c90** [comment]
Hi @reszelaz ,

I tried two different approaches following your workaround suggestions:

1. Creating two individual pseudo counter controller out of the same pseudo counter controller class, each one of them with only a single axis. Defining the first controller works, defining the second one fails with exactly the same error as reported in my initial post. It does not matter if both takes the same 2D detector as input or different ones. 

2. Duplicating the pseudo counter controller class and giving the second one a slightly different name (to make it fully independent). From each class, create a single axis pseudo counter controller with a different 2D detector assigned as input. Even here, creating the second one fails with the aforementioned error. 

Could it be, that generally there cannot be more than one 2D pseudo counter in total?

**Comment by reszelaz** [comment]
Hi Martin,
Thanks for trying the workarounds. Yes, they have the same problem.
I have investigated this bug and proposed a fix in #1446. For me your example works with these changes. Could you try it as well and comment back in the PR, please?
